hibernate-core/src/main/java/org/hibernate/mapping/Column.java

```diff
@@ -276,6 +276,17 @@ public String getSqlType(Dialect dialect, Mapping mapping) throws HibernateExcep
 		return sqlType;
 	}
 
+	private Type getType(Mapping mapping, Type type) {
+		while ( !( type instanceof JdbcMapping ) ) {
+			//ManyToOneType doesn't implement JdbcMapping
+			type = mapping.getIdentifierType( ( (EntityType) type ).getAssociatedEntityName() );
+			if ( type instanceof ComponentType ) {
+				type = ( (ComponentType) type ).getSubtypes()[getTypeIndex()];
```

Comment:
I saw there is some code duplication above (283-284 vs 251-252). Maybe we can eliminate it to avoid changing both when future refactoring is needed.